astropy/time/core.py

```diff
@@ -608,7 +610,10 @@ def strftime(self, format_spec):
             datetime_tuple = (sk['year'], sk['mon'], sk['day'],
                               sk['hour'], sk['min'], sk['sec'],
                               date_tuple[6], date_tuple[7], -1)
-            formatted_strings.append(strftime(format_spec, datetime_tuple))
+            if '%f' in format_spec:
+                format_spec = format_spec.replace('%f', str(sk['fracsec']))
```

Comment:
This permanently changes `format_spec` on the first iteration, then I think subsequent values get the same fracsec no matter what.

Reply:
This was a grave oversight from my part. I can't believe I wrote this and pushed. I think I was in a hurry to open the PR. Sorry.